source/base/mpi.cc

```diff
@@ -61,6 +61,23 @@ namespace Utilities
 
   namespace MPI
   {
+    void wait(const MPI_Comm &mpi_communicator)
+    {
+      // see https://www.open-mpi.org/faq/?category=debugging#serial-debuggers
+      std::cout << "PID " << ::getpid() << " on " << Utilities::System::get_hostname() << " ready for attach." << std::endl;
+
+      char a;
+      if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+        {
+          std::cout << "Type any character and press Enter/Return to start: " << std::flush;
+          std::cin >> a;
+        }
+
+#ifdef DEAL_II_WITH_MPI
+      MPI_Bcast(&a, 1, MPI_BYTE, 0, mpi_communicator);
```

Comment:
Why are usin `MPI_Bcast`? Shouldn't you use `MPI_Barrier`?

Reply:
agree, barrier should be enough.

Reply:
It doesn't actually make a difference. In a barrier, everyone has to wait till everyone else gets here. With the bcast, all but processor 0 have to wait, but processor 0 can continue as soon as it sends its data to all others, who are then free to continue once they receive the signal.